distributed/worker.py

```diff
@@ -82,6 +83,14 @@
     logger.warning("Please install psutil to estimate worker memory use")
     TOTAL_MEMORY = 8e9
     psutil = None
+try:
+    import resource
+    # minimum of physical available memory and the hard process limit
+    hard_limit = resource.getrlimit(resource.RLIMIT_RSS)[1]
+    if hard_limit > 0:
+        TOTAL_MEMORY = min(TOTAL_MEMORY, hard_limit)
+except OSError:
```

Comment:
When does this occur?

Also, should I be concerned about this behavior on OSX?  Presumably this would be handled by your use of `min`

```python
In [1]: import resource

In [2]: resource.getrlimit(resource.RLIMIT_RSS)[1]
Out[2]: 9223372036854775807
```

We also now strictly require psutil, so if you wanted to drop the`try-except` around the psutil import while you're here, that would be a welcome change :)

Reply:
Thanks for your feedback!

["An OSError is raised on syscall failure."](https://docs.python.org/3/library/resource.html) I assume that could occur on Windows/exotic OSs?

Maybe OSX uses this high number as a proxy for unlimited - what does `ulimit -m` return for you? On my linux machine it returns -1 for unlimited.

Ok, I will remove try try-catch around psutil.